pkg/autodiscovery/listeners/snmp.go

```diff
@@ -415,6 +417,19 @@ func (s *SNMPService) GetExtraConfig(key []byte) ([]byte, error) {
 		return []byte(s.config.Network), nil
 	case "loader":
 		return []byte(s.config.Loader), nil
+	case "tags":
+		return []byte(convertToCommaSepTags(s.config.Tags)), nil
 	}
 	return []byte{}, ErrNotSupported
 }
+
+func convertToCommaSepTags(tags []string) string {
```

Comment:
I'm not sure I understand this.

`tags` can be 
1) an array of `key:value` 
2) a string of comma-separated elements, such as `key1:value1,key2:value2`

This function is fine for case `1`, but I don't understand how it works for case `2`. Shouldn't we first split `s.config.Tags` on `,` ?

Reply:
@pducolin The config tags are like this: 

https://github.com/DataDog/datadog-agent/pull/7680/files#diff-54638e4af140aed843ac96c434105b678d4a402bea72e319fb05e0781f6ed459R203-R206

Then we replace `,` by `_` if needed.

Then the tags are joined with comma `,` and used in template.

An example:

If the `snmp_listener` config is like:

```
snmp_listener:
  configs:
    - network: 192.168.1.16/29
      version: 2
      port: 1161
      community: cisco_icm
      tags:
      - "tag1:val1"
      - "tag2:val2"
```

The Config.tags will be like `[]string{"tag1:val1", "tag2:val2"}` 

Then joined with `,` to be "tag1:val1,tag2:val2"

Then using the template, the result will be like:

```
    ip_address: 192.168.1.17
    ...
    extra_tags: "tag1:val1,tag2:val2"
```

Reply:
I see, the `snmp_listener` is passing the extra_tags configuration to the core_check as a yaml - thought it was passing an object